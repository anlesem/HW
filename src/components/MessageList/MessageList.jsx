import style from './MessageList.module.scss';

export const MessageList = (props) => {

   return (
      <ul className={style.messages}>Результат: {props.message.map((item, idx) => {
         return <li key={idx} className={style.message}>
            <p className={style.message_text}>{item.text}</p>
            <p className={style.message_author}>{item.author}</p>
         </li>;
      })}
      </ul>
   );
}