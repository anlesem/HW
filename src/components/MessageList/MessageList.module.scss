.message {
   margin-top: 20px;
   padding: 10px 20px;

   font-size: 14px;
   font-weight: normal;
   border-bottom: 1px solid #000;

   &_text {
      font-size: 16px;
   }

   &_author {
      margin-top: 20px;
      text-align: right;
   }

   &_robot {
      margin-top: 20px;
      text-align: right;
   }
}