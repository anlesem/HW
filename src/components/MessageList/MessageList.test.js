import { render, screen } from '@testing-library/react';
import { MessageList } from './MessageList';

describe('MessageList', () => {
  it('Компонент существует', () => {
    expect(MessageList).toBeInstanceOf(Function);
  });

  it('Снимок состояния', () => {
    const message = [
      {
        id: 1,
        author: 'somebody',
        text: 'some text'
      }
    ];
    const { asFragment } = render(<MessageList message={message} />);
    expect(asFragment(<MessageList />)).toMatchSnapshot();
  });

  it('Изначально список сообщений существует, но пустой', () => {
    const message = [];
    render(<MessageList message={message} />);

    expect(screen.getByRole('list')).toBeTruthy();
    expect(screen.queryByRole('listitem')).toBeFalsy();
  });

  it('Сформированный список сообщений', () => {
    const message = [
      {
        id: 1,
        author: 'somebody',
        text: 'some text'
      }
    ];
    render(<MessageList message={message} />);

    expect(screen.getByRole('listitem')).toBeTruthy();
    expect(screen.getByText(/somebody/)).toBeTruthy();
    expect(screen.getByText('some text')).toBeTruthy();
  });
});
