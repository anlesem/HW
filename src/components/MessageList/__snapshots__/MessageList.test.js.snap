// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList Снимок состояния 1`] = `
<DocumentFragment>
  <ul
    class="messages"
  >
    <li
      class="message"
      data-testid="list-item-0"
    >
      <p
        class="message_text"
      >
        some text
      </p>
      <p
        class="message_author"
      >
        somebody
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
